### DISCUSSION TITLE
Plausible Integration: Custom Domain support

### DISCUSSION THREAD

**Discussion by Pukimaa** [GitbookIO/community#9] [Feature Requests]
### What's your feature/idea?

I selfhost Plausible. However the Plausible Integration only uses the plausible.io script. I want that there is a setting to change the domain of the script and api/event.

### What's your desired outcome?

In the Plausible Integration Settings there is a new field called "Custom Script & API Domain" or something like that. If I put in `example.org` everything related to Plausible will be `https://example.org/script.js` and `https://example.org/api/event` instead of the `plausible.io` domain.

### What's the impact of this for you?

- [X] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Comment by SamyPesse** [comment]
Hey @Pukimaa 👋  We just released a new version of the integration with support for Self-hosting.

It adds a configuration to define the API endpoint. For security reasons, we will still load the script from `plausible.io`, but page views can be tracked on your self-hosted instance.

Let us know if that works!

**Comment by Pukimaa** [comment]
Hey @SamyPesse 👋

How exactly does it work? I added the domain I use for the Docs, but it's not showing up any analytics when visiting my docs.

So Plausible sits at stats.pukima.site and my docs at docs.tako-bot.com so I set the field of the config to docs.tako-bot.com

Isn't that how it works?

**Reply by SamyPesse** [reply]
If Plausible sits at `https://stats.pukima.site`, then you need to configure `https://stats.pukima.site/api/event` as "Self-hosted API".

GitBook will make an HTTP request to `https://stats.pukima.site/api/event` following the "POST /api/event" spec (https://plausible.io/docs/events-api#post-apievent) for each page view.

For the "Domain", when self-hosting, I don't know exactly what you should configure, but it's probably the identifier of the website instance that you created in your Plausible instance.

**Comment by acejam** [comment]
Plausible now issues you a personalized site-specific JS snippet URL. Many of the previous generic ones (script.js) are blocked now. This means users are missing Plausible traffic analytics when using GitBook.

@SamyPesse An additional field that allows you to load a specific JS snippet would resolve this.
